### PR TITLE
Migrate to a different (new) skeleton

### LINKED ISSUE #50: Re-skeletonize from the `cisagov/skeleton-aws-lambda-python` project
# 💡 Summary #

This project should be re-skeletonized to descend from the [cisagov/skeleton-aws-lambda-python] project.

## Motivation and context ##

Currently this project is based on the [cisagov/skeleton-aws-lambda] project. However that project has been deprecated in favor of the [cisagov/skeleton-aws-lambda-python] project. This new project represents a much more modern approach that can also leverage more functionality that AWS has added to Lambdas.

## Implementation notes ##

The structure of the two skeleton projects is completely different. Re-skeletonizing this project will involve dismantling the Python package currently defined in this project. Since the Python package is not particularly complex it should be straightforward to refactor.

## Acceptance criteria ##

- [ ] Repository is updated to be based on [cisagov/skeleton-aws-lambda-python] project.

[cisagov/skeleton-aws-lambda]: https://github.com/cisagov/skeleton-aws-lambda
[cisagov/skeleton-aws-lambda-python]: https://github.com/cisagov/skeleton-aws-lambda-python

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request migrates this repository's upstream skeleton from [cisagov/skeleton-aws-lambda] to [cisagov/skeleton-aws-lambda-python]. This closes #50.

## 💭 Motivation and context ##

The [cisagov/skeleton-aws-lambda] project has been archived in favor of the new [cisagov/skeleton-aws-lambda-python] skeleton. This skeleton better aligns with new options for building lambdas from AWS and reorganizes Python lambda projects to be about building deployment artifacts vs housing Python packages that also get built into deployment artifacts. As a result downstream repositories like this one must be updated.

### ⚠ Note ###

This is essentially a lift-and-shift migration. This project still needs a rework once this pull request is merged to better align the existing functionality with the new skeleton to truly be "finished".

## 🧪 Testing ##

Automated tests pass. I verified that the generated build artifact correctly processed data in my testing environment.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Add a tag or create a release.

[cisagov/skeleton-aws-lambda]: https://github.com/cisagov/skeleton-aws-lambda
[cisagov/skeleton-aws-lambda-python]: https://github.com/cisagov/skeleton-aws-lambda-python

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

**Comment by mcdonnnj** [comment]
The CodeQL tests are stuck because the workflow was disabled due to inactivity. They should kick off when I bump or finalize the version now that I've re-enabled the workflow.

**Comment by dav3r** [comment]
> This is essentially a lift-and-shift migration. This project still needs a rework once this pull request is merged to better align the existing functionality with the new skeleton to truly be "finished".

Is there an issue (or issues) documenting what is needed here?

**Review by dav3r — APPROVED** [review]
Looks great- nice work! 👍

**Comment by mcdonnnj** [comment]
> > This is essentially a lift-and-shift migration. This project still needs a rework once this pull request is merged to better align the existing functionality with the new skeleton to truly be "finished".
> 
> Is there an issue (or issues) documenting what is needed here?

I was waiting for any requested changes before creating the issue so I could be explicit in what I was detailing.

File: README.md [hunk 1]
```diff
@@ -41,20 +27,74 @@ blank, the script will remove that JSON element from the record.
...
+
+## Running the Lambda locally ##
 
-Please note that the corresponding Docker image _must_ be rebuilt
-locally if the script `build.sh` changes.  Given that rebuilding the Docker
-image is very fast (due to Docker's caching) if the script has not changed, it
-is a very good idea to _always_ run the `docker-compose build` step when
-using this tool.
+The configuration in this repository allows you run the Lambda locally for
+testing as long as you do not need explicit permissions for other AWS
```
Comment (jsf9k) [inline README.md:62]: I think this word is slightly better here, but please disregard if you disagree.
```suggestion
testing as long as you do not require explicit permissions for other AWS
```

**Review by jsf9k — APPROVED** [review]
I made one very minor suggestion, but this PR is approved even if you choose to disregard it.

File: README.md (hunk 1 above)
Comment (mcdonnnj) [reply README.md:62]: This is pulled in from the skeleton so I will punt the change upstream.

**Review by dav3r — APPROVED** [review]
🎯

**Pull request merged**